presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/RuleAssert.java

```diff
@@ -81,6 +88,15 @@ public RuleAssert withSession(Session session)
         return this;
     }
 
+    public RuleAssert withBefore(Set<Rule<?>> rules)
```

Comment:
The order (`before`) here is not necessary. Tested rule should not fire before any traits are computed, so the explicit ordering is not needed. At very most we could just assert that trait rules executed before tested rule.